utils/gen_config.py

```diff
@@ -0,0 +1,174 @@
+    """Generate list of hosts from profile and report data.
+
+    Args:
+        profilefile (string path to file):
+            Expects csv with columns:
+                - name
+                - ip
+                - profile
+                - auth
+        reportfile (string path to file):
+            Expects csv from rho scan
+
+    .. note::
+        It will generated all new hosts, old host data is discarded.
+    """
+    hosts = []
+    with open(profilefile) as csvfile:
+        hostdata = csv.DictReader(csvfile)
+        for row in hostdata:
+            hosts.append({
+                'hostname': row.get('name', ''),
+                'ip': row.get('ip', ''),
+                'provider': 'vcenter' if 'vc' in row.get('name', '') else '',
+                'facts': {},
+                'profile': row.get('profile', uuid.uuid4()),
+                'auth': row.get('auth', ''),
+            })
+    with open(reportfile) as csvfile:
+        scan_results = csv.DictReader(csvfile)
+        for row in scan_results:
+            for host in hosts:
+                if host['ip'] == row.get('connection.host', ''):
+                    for key in row.keys():
+                        if 'date' not in key:
+                            host['facts'][key] = row[key]
+    return hosts
+
+
+def gen_profiles(cfg, hosts):
+    """Generate profiles for given config and hosts.
+
+    Args:
+        cfg (dictionary):
+            Expects valid camayoc config file
+
+        hosts (list):
+            Expects output of gen_hosts()
+
+    If hosts have profiles associated with them, they will be added to
+    appropriate profile.
+
+    If the profile exists, they will be added to it. If it does not,
+    a new profile is generated.
+
+    If the profile does not contain the auth matching the host, the
+    auth will be added.
+
+    .. note::
+        This is "constructive", as in it adds profiles and modifies profiles,
+        but does not delete existing profiles.
+    """
+    profiles = cfg.get('profiles', [])
+    for host in hosts:
+        profile_found = False
+        # first append hosts to any existing profiles
+        for profile in profiles:
+            if host.get('profile', '') == \
+                    profile.get('name', uuid.uuid4()):
+                profile_found = True
+                if host.get('ip', '') not in profile.get('hosts', []):
+                    if host['ip'] != '':
+                        profile['hosts'].append(host['ip'])
+                if host['auth'] not in profile.get('auths', []):
+                    if host['auth'] != '':
+                        profile['auths'].append(host['auth'])
+        if not profile_found:
+            profiles.append({
+                'name': host['profile'],
+                'hosts': [host['ip']],
+                'auths': [host['auth']],
+            })
+    return profiles
+
+
+def write_config(cfg, hosts, profiles, configfile):
+    """Write changes to camayoc config file."""
```

Comment:
To follow the documentation pattern form the script, add the documentation for the parameters.